src/Home.js

```diff
@@ -0,0 +1,351 @@
+                ))}
+              </div>
+              {/* Dates of the Week */}
+              <div className='daysOfWeek'>
+                {currentDates.map((date, index) => (
+                  <div key={index}><h5 className='dateStyle'>{date.getDate().toString().padStart(2, '0')}</h5></div>    
+                ))}
+              </div>
+          </div>
+          <ForwardsButton startDate={startDate} setStartDate={setStartDate}/>
+        </nav>
+      </div>
+  
+      
+    )
+  }
+  // display back button
+  const BackButton = ({startDate, setStartDate}) => {
+    const handlePrevWeek = () => {
+      const newStartDate = new Date(startDate);
+      newStartDate.setDate(startDate.getDate() - 7);
+      setStartDate(newStartDate);
+    }
+  
+    return (
+      <button className='calendarNavButton' onClick={handlePrevWeek}>
+        <img src='https://cdn-icons-png.flaticon.com/128/2989/2989988.png' 
+        alt='previous' 
+        style={{width: 25, 
+                height: 'auto', 
+                transform: 'scaleX(-1)', 
+                filter: 'invert(29%) sepia(46%) saturate(312%) hue-rotate(212deg) brightness(98%) contrast(89%)'}} />
+      </button>
+    )
+  }
+  // display forwards button
+  const ForwardsButton = ({startDate, setStartDate}) => {
+    const handleNextWeek = () => {
+      const newStartDate = new Date(startDate);
+      newStartDate.setDate(startDate.getDate() + 7);
+      setStartDate(newStartDate);
+    }
+  
+    return (
+      <button className='calendarNavButton' onClick={handleNextWeek}>
+        <img src='https://cdn-icons-png.flaticon.com/128/2989/2989988.png' 
+        alt='next' 
+        style={{width: 25, 
+                height: 'auto', 
+                filter: 'invert(29%) sepia(46%) saturate(312%) hue-rotate(212deg) brightness(98%) contrast(89%)'}} />
+      </button>
+    )
+  }
+  
+  // displays timestamps for calendar side display
+  const Times = () => {
+    return (
+      <section className='timeMarkings'>
+        {TIMES.map(time => (
+          <h5 className='timeStyle' key={time}>{time}</h5>
+        ))}
+      </section>
+    )
+  }
+  // stores all timestamps for calendar side display
+  const TIMES = [
+    '7 AM', '8 AM', '9 AM', '10 AM', '11 AM', '12 PM', 
+    '1 PM', '2 PM', '3 PM', '4 PM', '5 PM', '6 PM', 
+    '7 PM', '8 PM', '9 PM'
+  ]
+  
+  // the following 2 functions determines grid placement for a time
+  const getGridCol = (date) => {
+    switch (date.getDay()) {
+      case 0:
+        return 'sunStart';
+      case 1:
+        return 'monStart';
+      case 2:
+        return 'tueStart';
+      case 3:
+        return 'wedStart';
+      case 4:
+        return 'thuStart';
+      case 5:
+        return 'friStart';
+      case 6:
+        return 'satStart';
+      default:
+        return;
+    }
+  }
+  const getGridRow = (date) => {
+    switch (date.toLocaleString('default', {hour: '2-digit', minute: '2-digit'})) {
+      case '07:00 AM':
```

Comment:
Try to avoid making large maps like this whenever possible. 

The purpose of this map is to map the time to a grid row. You can do this more easily with a mathematical function. the JavaScript Date object provides a `.getHours()` function and a `.getMinutes()` function. 
Say row 1 corresponds to an hour = 7 and minutes = 0. Row 2 corresponds to hour = 7 and a minutes = 30. Row 3 is hour = 8 and minutes = 0.

Given any hour H and any minutes M, you can find the row using ((H-7)*2) + (1*M).
- We do H-7 because the first row, row 0, is 7am.
- We do (H-7)*2 because every hour takes up 2 rows. So a +1 increase in hour is a +2 increase in rows. This gives us row 0 for 7:xx, row 2 for 8:xx, row 4 for 9:xx, etc.
- We do +(1*M) because this will add 1 to our row if M is nonzero (which happens when M is 30), otherwise it won't add 1. This has the effect of adding 1 to our row if we are on the half hour. So 8:30 = row 2 + 1, while 8:00 = row 2.

This works with the `repeat` function i suggested you use when defining your grid-template-rows and grid-template-columns.

Reply:
Ok i forgot that rows are actually 1-index instead of 0-indexed, so you will have to add 1 to the final result. 

[This guide](https://css-tricks.com/snippets/css/complete-guide-grid/) should help you out